sprocs/sync_groups.sql

```diff
@@ -0,0 +1,95 @@
+-- Accepts a course instance ID and a JSON object specifying the, as an object of
```

Comment:
I'm going to hold off reviewing this for now. We'll want to compare it against how the other sync functions are written, and make sure we're not in any race or locking conditions with multiple course instances getting sync'd. 

Also, from our slack chats, we need to decide on a strategy for group deletion and what that looks like. It kinda relates to the group schema and how we're using `group_configs`. I'm thinking we probably want to make a group_config entry for "loaded from infoCourseInstance.json" and use that relationship to control what this process does for its groups.

Reply:
I'm still in favor of "loaded from infoCourseInstance.json" as being implicitly associated with a null value in group_configs, since configs are built specifically for being associated with assessments. If we don't go with that, I'd still rather use an enum field in groups rather than an arbitrary text field inside of a config.